backends/jny/jny.cc

```diff
@@ -0,0 +1,554 @@
+        }
+
+        if (_include_attributes) {
+            f << ",\n" << _indent << "    \"attributes\": {\n";
+
+            write_prams(cell->attributes, indent_level + 2);
+
+            f << "\n";
+            f << _indent << "    }";
+        }
+
+        if (_include_properties) {
+            f << ",\n" << _indent << "    \"parameters\": {\n";
+
+            write_prams(cell->parameters, indent_level + 2);
+
+            f << "\n";
+            f << _indent << "    }";
+        }
+
+        f << "\n" << _indent << "  }";
+    }
+};
+
+struct JnyBackend : public Backend {
+    JnyBackend() : Backend("jny", "generate design metadata") { }
+    void help() override {
+        // XXX(aki): TODO: explicitly document the JSON schema
+        //   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+        log("\n");
+        log("    jny [options] [selection]\n");
+        log("\n");
+        log("    -connections\n");
+        log("        Include connection information in the netlist output.\n");
+        log("\n");
+        log("    -attributes\n");
+        log("        Include attributed information in the netlist output.\n");
+        log("\n");
+        log("    -properties\n");
+        log("        Include property information in the netlist output.\n");
+        log("\n");
+        log("Write a JSON metadata for the current design\n");
+        log("\n");
+        log("\n");
+    }
+
+    void execute(std::ostream *&f, std::string filename, std::vector<std::string> args, RTLIL::Design *design) override {
+
+        bool connections{false};
+        bool attributes{false};
+        bool properties{false};
+
+        size_t argidx{1};
+        for (; argidx < args.size(); argidx++) {
+            if (args[argidx] == "-connections") {
+                connections = true;
+                continue;
+            }
+
+            if (args[argidx] == "-attributes") {
+                attributes = true;
+                continue;
+            }
+
+            if (args[argidx] == "-properties") {
+                properties = true;
+                continue;
+            }
+
+            break;
+        }
+        extra_args(f, filename, args, argidx);
+
+        log_header(design, "Executing jny backend.\n");
+
+        JnyWriter jny_writer(*f, false, connections, attributes, properties);
+        jny_writer.write_metadata(design);
+    }
+
+} JnyBackend;
+
+
+struct JnyPass : public Pass {
+    JnyPass() : Pass("jny", "write design and metadata") { }
+
+    void help() override {
+        //   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+        log("\n");
+        log("    jny [options] [selection]\n");
+        log("\n");
+        log("Write a JSON netlist metadata for the current design\n");
+        log("\n");
+        log("    -o <filename>\n");
+        log("        write to the specified file.\n");
+        log("\n");
+        log("    -connections\n");
```

Comment:
swap the defaults and by default write all information

Reply:
Switched all the defaults, so now you need to explicitly disable the output of the connections, attributes, and properties